Sources/JWTKit/JWTError.swift

```diff
@@ -1,44 +1,224 @@
+            case claimVerificationFailure
+            case signingAlgorithmFailure
+            case malformedToken
+            case signatureVerifictionFailed
+            case missingKIDHeader
+            case unknownKID
+            case invalidJWK
+            case invalidBool
+            case generic
+        }
+        
+        internal let base: Base
+        
+        private init(_ base: Base) {
+            self.base = base
+        }
+        
+        public static let claimVerificationFailure = Self(.claimVerificationFailure)
+        public static let signingAlgorithmFailure = Self(.signingAlgorithmFailure)
+        public static let malformedToken = Self(.malformedToken)
+        public static let signatureVerificationFailed = Self(.signatureVerifictionFailed)
+        public static let missingKIDHeader = Self(.missingKIDHeader)
+        public static let unknownKID = Self(.unknownKID)
+        public static let invalidJWK = Self(.invalidJWK)
+        public static let invalidBool = Self(.invalidBool)
+        public static let generic = Self(.generic)
+        
+        public var description: String {
+            switch self.base {
+            case .claimVerificationFailure:
+                "claimVerificationFailure"
+            case .signingAlgorithmFailure:
+                "signingAlgorithmFailure"
+            case .malformedToken:
+                "malformedToken"
+            case .signatureVerifictionFailed:
+                "signatureVerifictionFailed"
+            case .missingKIDHeader:
+                "missingKIDHeader"
+            case .unknownKID:
+                "unknownKID"
+            case .invalidJWK:
+                "invalidJWK"
+            case .invalidBool:
+                "invalidBool"
+            case .generic:
+                "generic"
+            }
         }
     }
-
-    public var description: String {
-        return "JWTKit error: \(self.reason)"
+    
+    private final class Backing {
+        fileprivate var errorType: ErrorType
+        fileprivate var name: String?
+        fileprivate var reason: String?
+        fileprivate var underlying: Error?
+        fileprivate var kid: JWKIdentifier?
+        fileprivate var identifier: String?
+        fileprivate var failedClaim: (any JWTClaim)?
+        
+        init(errorType: ErrorType) {
+            self.errorType = errorType
+        }
+        
+        func copy() -> Self {
+            let new = Self.init(errorType: self.errorType)
+            new.name = self.name
+            new.reason = self.reason
+            new.underlying = self.underlying
+            new.kid = self.kid
+            new.identifier = self.identifier
+            return new
+        }
+    }
+    
+    private var backing: Backing
+    
+    private mutating func copyBackingStorageIfNecessary() {
+        if !isKnownUniquelyReferenced(&self.backing) {
+            self.backing = self.backing.copy()
+        }
+    }
```

Comment:
I don't think you really need this optimization here. Fabian used it on `PSQLError` because some of the associated values are more than a little nontrivial, but the only really potentially sizable field of this type is the `failedClaim`, and that'd be rare as well.